src/App.tsx

```diff
@@ -1,24 +1,129 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { createTodo, deleteTodo, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Error } from './components/Error';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { useSort } from './utils/useSort';
+import { Sort } from './types/Sort';
 
-const USER_ID = 0;
+const USER_ID = 6771;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorType, setErrorType] = useState<string>('');
+  const [sortBy, setSortBy] = useState<Sort>(Sort.all);
+  const [isInputDisabled, setIsInputDisabled] = useState<boolean>(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodo, setLoadingTodo] = useState([0]);
+  const isError = !!errorType || errorType !== '';
+
+  async function fetchTodos() {
+    setErrorType('');
+    try {
+      const todosData = await getTodos(USER_ID);
+
+      setTodos(todosData);
+    } catch {
+      setErrorType('Error loading data');
+    }
+  }
+
+  useEffect(() => {
+    fetchTodos();
+  }, []);
+
+  const sortedTodos = useSort(todos, sortBy);
+
+  const addTodo = async (title: string) => {
+    if (!title.trim()) {
+      setErrorType('Title can\'t be empty');
+
+      return;
+    }
+
+    setIsInputDisabled(true);
+
+    try {
+      const newTodo = await createTodo(USER_ID, {
+        title,
+        userId: USER_ID,
+        completed: true,
+      });
+
+      setTempTodo({ ...newTodo, id: 0 });
+
+      setTodos([...todos, newTodo]);
+      setTempTodo(null);
```

Comment:
the main idea is to show todo during this creating, but you set it after creating the todo. fix that plz